File: baekjoon/gold/11401/11401.cpp
#include <iostream>

const long long P = 1000000007;

long long factorial(long long n);
long long pow(long long base, long long exp);

int main() {
  long long n, k, denom;

  scanf("%lld %lld\n", &n, &k);
  denom = (factorial(k) * factorial(n - k)) % P;
  printf("%lld\n", (factorial(n) * pow(denom, P - 2)) % P);
}

long long factorial(long long n) {
    long long result = 1;
    for (long long i = 1; i <= n; i++) {
        result *= i;
        result %= P;
    }
    return result;
}

long long pow(long long base, long long exp) {
  long long ret = 1;
  for (long long i = exp; i > 0; i /= 2) {
    if (i % 2 == 1) {
      ret *= base;
      ret %= P;
    }
    base = (base * base) % P;
  }
  return ret;
}

File: baekjoon/gold/11401/11401.java
import java.util.Scanner;

class Main {
  static long P = 1000000007;
  public static void main(String[] args) {
    long n, k, denom;
    Scanner sc = new Scanner(System.in);
    n = sc.nextLong();
    k = sc.nextLong();

    denom = (factorial(k) * factorial(n - k)) % P;
    System.out.println((factorial(n) * pow(denom, P - 2)) % P);
  }

  public static long factorial(long n) {
    long ret = 1;
    for (long i = 1; i <= n; i++) {
      ret *= i;
      ret %= P;
    }
    return ret;
  }

  public static long pow(long base, long exp) {
    long ret = 1;
    for (long i = exp; i > 0; i /= 2) {
      if (i % 2 == 1) {
        ret *= base;
        ret %= P;
      }
      base *= base;
      base %= P;
    }
    return ret;
  }
}